test/DepositTest.t.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "./ZeroState.t.sol";
+import "../lib/forge-std/src/Test.sol";
+
+contract DepositTest is ZeroState {
+
+    event Deposit(address indexed user, uint256 amount, uint256 balance);
+
+    function testDepositBelowAllowance() public {
```

Comment:
You just need to test the deposit, not that the allowance mechanism in ERC20 works.
```suggestion
    function testDepositInitialBalance() public {
```